ReactiveCocoa/Swift/Signal.swift

```diff
@@ -829,7 +829,7 @@ extension SignalType {
 	/// multiple times) by `sampler`, then complete once both input signals have
 	/// completed, or interrupt if either input signal is interrupted.
 	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
-	public func sampleOn(sampler: Signal<(), NoError>) -> Signal<Value, Error> {
+	public func sampleOn<S: SignalType where S.Error == NoError>(sampler: S) -> Signal<Value, Error> {
```

Comment:
This is on purpose: the current signature forces users to explicitly ignore values rather than that happening transparently.